libqtile/lazy.py

```diff
@@ -47,7 +47,7 @@ def __init__(self, call: CommandGraphCall, args: Tuple, kwargs: Dict) -> None:
         self._kwargs = kwargs
 
         self._layouts: Set[str] = set()
-        self._when_floating = True
+        self._when_floating = None
```

Comment:
```suggestion
        self._when_floating: Optional[bool] = None
```
This is the initial declaration of `when_floating` so you need to declare the type here.

This should fix the mypy error you're seeing.